tsdb/index/inmem/inmem.go

```diff
@@ -701,10 +701,13 @@ type ShardIndex struct {
 
 // CreateSeriesListIfNotExists creates a list of series if they doesn't exist in bulk.
 func (idx *ShardIndex) CreateSeriesListIfNotExists(keys, names [][]byte, tagsSlice []models.Tags) error {
-	// Ensure that no tags go over the maximum cardinality.
 	var reason string
-	var dropped, n int
+	var dropped int
+
+	// Ensure that no tags go over the maximum cardinality.
 	if maxValuesPerTag := idx.opt.Config.MaxValuesPerTag; maxValuesPerTag > 0 {
+		var dropped, n int
```

Comment:
This `dropped` is shadowing the `dropped` above. It will never be non-zero for this `if`: https://github.com/influxdata/influxdb/blob/3b25cffda4a1e2b05a18ebb42a95cf18fa4b2412/tsdb/index/inmem/inmem.go#L748-L754